### DISCUSSION TITLE
Use .snupkg instead of embedding the pdb in the .nupkg

### DISCUSSION THREAD

**Discussion by obligaron** [reqnroll/Reqnroll#147] [Ideas]
Currently a user of Reqnroll can't debug into Reqnroll itself out of the box.
He has to set `<CopyDebugSymbolFilesFromPackages>true</CopyDebugSymbolFilesFromPackages>` in his csproj.

We would create a explicit [NuGet symbol packages](https://learn.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg). This would allow a user to debug into the project without any additional steps.

Side note: We should consider enabling [deterministic builds](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/code-generation#deterministic) when publishing a new version. This is checked by [nuget.info](https://nuget.info/packages/Reqnroll/2.0.0).

**Comment by gasparnagy** [comment]
I'm for it.

And also when we touch the NuGet packages, we should set repo URL, like @thompson-tomo suggested for SpecFlow.

**Reply by ajeckmans** [reply]
That is already automatically done if you build using .net 8 SDK.
See https://github.com/dotnet/sourcelink

So I'd check the produced packages first before you start hardcoding these.

**Reply by gasparnagy** [reply]
Then we do something wrong. Because we built v2 with .NET 8, but the source link is not there on nuget: https://www.nuget.org/packages/Reqnroll

**Reply by ajeckmans** [reply]
No, 
![image](https://github.com/reqnroll/Reqnroll/assets/2088777/c4e16589-b307-4c0f-b2be-be43d415413e)
The issue is the snupkg is not uploaded. Once you do that then nuget.org will also show the sourcelink stuff. See also the bottom of the first comment: https://github.com/dotnet/docfx/issues/8900

**Reply by gasparnagy** [reply]
PR: #161

**Comment by gasparnagy** [comment]
This is fixed in v2.0.3 hopefully
